binder/requirements.txt

```
@@ -0,0 +1 @@
+hydra-core==1.0.0rc2
```

maybe this will work?
```suggestion
--pre
hydra-core
```

---

This throws some error when building the binder notebook

```
Waiting for build to start...
Picked Git content provider.
Cloning into '/tmp/repo2docker1ts59ep7'...
HEAD is now at 73209e3f Incorporate feedback.
Error during build: list index out of range
```

Will check with binder team.

---

I am starting to think that a docker image might be the better choice for the future.
Mixing code from the repo with the latest version is a sure way to have issues later.
This same docker image will also be needed to address the exact same issue with OmegaConf once it has it's own grammar.

---

Do we care about the time to build the docker image? I understand that even now, binder makes a docker file. In my experience, installing the version from pip was quite fast. I think the startup time will be higher if we have to build the code (because of the additional dependencies).

---

build what code?
installing jre should be pretty fast.

---

Installing conda, installing jre, intalling all dependencies... vs installing one python package (ie hydra)

---

well, we just need to add jre to whatever was in the preview docker image.

---

It seems docker is not the recommended mode for binder [0]. Quoting from the link:

```
We recommend against using a Dockerfile as a way to make your repository usable with binder. Use them as a last resort after all methods in Preparing a repository for Binder have failed.

Binder’s requirements for Dockerfiles are in beta and subject to change. Dockerfiles may break on Binder from time to time during the beta period.

``` 

[0]: https://mybinder.readthedocs.io/en/latest/tutorials/dockerfile.html

---

Can we ask them what is their recommendation for getting jre on a docker image? This seems like a very fundamental tool and they can just include it with the default image.

---

just realized that they support installing via apt-get as well https://mybinder.readthedocs.io/en/latest/config_files.html#apt-txt-install-packages-with-apt-get